src/test/java/org/folio/rest/support/clients/CqlQuery.java

```diff
@@ -0,0 +1,45 @@
+package org.folio.rest.support.clients;
+
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+
+public final class CqlQuery {
+  private final String query;
+
+  public CqlQuery(String query) {
+    this.query = query;
+  }
+
+  public static CqlQuery exactMatch(String propertyName, Object value) {
+    return fromTemplate("%s==\"%s\"", propertyName, value);
+  }
+
+  public static CqlQuery startsWith(String propertyName, Object value) {
+    return fromTemplate("%s=\"%s\"", propertyName, value);
+  }
+
+  public static CqlQuery lessThen(String propertyName, Object bound) {
+    return fromTemplate("%s<\"%s\"", propertyName, bound);
+  }
+
+  public static CqlQuery fromTemplate(String template, Object... values) {
+    final Object[] maskedValues = Stream.of(values)
+      .map(CqlQuery::maskCqlValue).toArray();
+
+    return new CqlQuery(String.format(template, maskedValues));
+  }
+
+  public final String asString() {
+    return query;
+  }
+
+  private static Object maskCqlValue(Object value) {
+    if (!(value instanceof String) || StringUtils.isBlank((String) value)) {
+      return value;
+    }
+
+    return value.toString().replace("\\", "\\\\")
+      .replace("\"", "\\\"");
+  }
```

Comment:
Can you mask all cases? Look at these tests:

```
package org.folio.rest.support.clients;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

import org.junit.Test;

public final class CqlQueryTest {
  @Test
  public void backslashString() {
    assertThat(CqlQuery.exactMatch("foo", "\\").asString(), is("foo==\"\\\\\""));
  }

  @Test
  public void backslashChar() {
    assertThat(CqlQuery.exactMatch("foo", '\\').asString(), is("foo==\"\\\\\""));
  }

  @Test
  public void backslashClass() {
    class MyClass {
      public String toString() {
        return "\\";
      }
    }
    assertThat(CqlQuery.exactMatch("foo", new MyClass()).asString(), is("foo==\"\\\\\""));
  }

  // If %c is supported is must properly mask.
  // Removing support for %c is fine.
  @Test
  public void charFromInt() {
    int c = '\\';
    assertThat(CqlQuery.fromTemplate("foo==\"%c\"", c).asString(), is("foo==\"\\\\\""));
  }
}
```

My suggestion is to remove support for %c and do it this way:

```
  private static String maskCqlValue(Object value) {
    return value.toString()
        .replace("\\", "\\\\")
        .replace("\"", "\\\"")
        .replace("*", "\\*")
        .replace("?", "\\?")
        .replace("^", "\\^");
  }
```

This also masks * ? ^ that are special CQL characters: https://www.loc.gov/standards/sru/cql/contextSets/theCqlContextSet.html

Reply:
Sure, modified.